src/sqlfluff/dialects/dialect_spark3.py

```diff
@@ -1515,6 +1515,35 @@ class SamplingExpressionSegment(BaseSegment):
     )
 
 
+@spark3_dialect.segment()
+class LateralViewClauseSegment(BaseSegment):
+    """A `LATERAL VIEW` like in a `FROM` clause.
+
+    https://spark.apache.org/docs/latest/sql-ref-syntax-qry-select-lateral-view.html
+    """
+
+    type = "lateral_view_clause"
+
+    match_grammar = Sequence(
+        Indent,
+        "LATERAL",
+        "VIEW",
+        Ref.keyword("OUTER", optional=True),
+        Ref("FunctionSegment"),
+        # NB: AliasExpressionSegment is not used here for table
+        # or column alias because `AS` is optional within it
+        # (and in most scenarios). Here it's explicitly defined
+        # for when it is required and not allowed.
+        Ref("SingleIdentifierGrammar", optional=True),
+        Sequence(
+            "AS",
+            Ref("SingleIdentifierGrammar"),
+            optional=True,
+        ),
```

Comment:
My read of [the above link](https://spark.apache.org/docs/latest/sql-ref-syntax-qry-select-lateral-view.html) is this should be:

```suggestion
        Sequence(
            "AS",
            Delimited(
                Ref("SingleIdentifierGrammar")
            ),
            optional=True,
        ),
```

Is that so? And if so should we add a test case for multiple aliases?